src/components/new-safe/CreateSafe/useCreateSafe.ts

```diff
@@ -1,12 +1,30 @@
 import useIsWrongChain from '@/hooks/useIsWrongChain'
 import useWallet from '@/hooks/wallets/useWallet'
+import { useEffect } from 'react'
+import useLocalStorage from '@/services/local-storage/useLocalStorage'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import type { PendingSafeData } from '@/components/new-safe/steps/Step4'
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe/index'
+import { SAFE_PENDING_CREATION_STORAGE_KEY } from '@/components/new-safe/steps/Step4'
 
-const useCreateSafe = () => {
+const useCreateSafe = (setStep: StepRenderProps<NewSafeFormData>['setStep']) => {
+  const [pendingSafe] = useLocalStorage<PendingSafeData | undefined>(SAFE_PENDING_CREATION_STORAGE_KEY)
   const wallet = useWallet()
   const isWrongChain = useIsWrongChain()
 
   const isConnected = wallet && !isWrongChain
 
+  useEffect(() => {
+    if (isWrongChain || !wallet) {
+      setStep(0)
+    }
+
+    // Jump to the status screen if there is already a tx submitted
```

Comment:
I tried `useLayoutEffect` but still get the initial flicker. Another thing I looked into was setting the `initialStep` depending on `pendingSafe` but it also doesn't work because our `useLocalStorage` is not render blocking.

Reply:
Alright, let's not worry about it.